### PR TITLE
Adds happiness (a new chemical)

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

Adds the happiness chemical, a drug. Its description is "Fills you with ecstasic numbness and causes minor brain damage. If overdosed causes sudden mood swings and spikes in heart rate."<br><br>

It can be found in the psych's office, and maint pill bottles. It can also be made in chemistry and grown from rainbow weed in botany. Its primarily purpose is to be a silly chemical, but it also removes some confusion and some jitter (TG, where I'm spiritually porting this from, has it remove ALL confusion, which felt a little strong). Right now, it has the same amount of confusion removal as degreaser for IPCs. Any organic who takes it, will laugh, giggle, and grin their heart out. Along with hearing some messages about feeling good. The chemical deals a small amount of brain damage, but can be outhealed by mannitol, which is also found in the psych's pills. It causes minor pain numbing effects, but the least out of any other painkiller. (20 shock_reduction, 1% surgery improvement)<br>

However, the fun part starts when someone overdoses on it, causing them go to into a "mood swing" rapidly switching affirmations and well... non-affirmations, as well as making them slightly dizzy. It also makes the person switch between laughing and smiling to crying and sulking randomly. This is pretty comedic to see it taking up your chat. When overdosing, it will also increase your pulse slightly, and cause more brain damage.

Its recipe is (1 space drugs, 1 nitrogen, 1 oxygen, 2 sulphuric acid) and requires a catalyst of (5 plasma). Its a fun recipe to learn and make, while not being too difficult for a minor and fun drug.

The messages are subject to change if people feel they don't fit well.

## Why It's Good For The Game
More psych/chemistry content is cool, especially if it has RP content outside of sec vs antag. I could totally see someone trying to run a drug smuggling ring with these, because they're not to bad to actually take.

## Images of changes

Some happy floor pills, and a beaker full of happiness.
image
An example of overdosing as much as you can.
image
Taking the pills normally, no overdosing
image

## Testing

See above, tested out making the chemical, finding it in bottles, etc.

## Changelog
:cl:
add: A new chemical, "happiness" a mood stabilizer until overdose, where it then causes mood swings instead
/:cl:

### CODE REVIEW

**Comment by pismorizzo** [comment]
Didn't name it Joy. 0/10...

**Comment by SynthTwo** [comment]
The overdosing is fucking hilarious. WE'RE SO BACK!!!!!!!

**Comment by Pennwick** [comment]
Is there a reason it uses a plasma catalyst? I believe aside from condensed capsaicin is the only chem that requires one and thats because its just a reaction with itself.

**Comment by Contrabang** [comment]
> Is there a reason it uses a plasma catalyst? I believe aside from condensed capsaicin is the only chem that requires one and thats because its just a reaction with itself.

Just to make it more interesting to make, I believe happiness on TG also has the same plasma catalyst for it

File: code/game/objects/items/contraband.dm [hunk 1]
```diff
@@ -3,12 +3,13 @@
 //Illicit drugs~
 /obj/item/storage/pill_bottle/happy
 	name = "Happy pills"
-	desc = "Highly illegal drug. When you want to see the rainbow."
+	desc = "Real fun drugs, for when you want to see the rainbow. Happy happy joy joy!"
```
Comment (JimKil3) [inline code/game/objects/items/contraband.dm:6]:
```suggestion
	desc = "Real fun drugs for when you want to see the rainbow. Happy happy joy joy!"
```
this is pretty nitpicky, but I don't think this comma is neccessary

File: code/modules/reagents/chemistry/reagents_datum.dm [hunk 2]
```diff
@@ -287,3 +287,6 @@
 	if(M.healthdoll)
 		M.healthdoll.cached_healthdoll_overlays.Cut()
 	M.updatehealth("fakedeath reagent end")
+
+/datum/reagent/proc/has_heart_rate_increase()
+	return heart_rate_increase
```
Comment (JimKil3) [inline code/modules/reagents/chemistry/reagents_datum.dm:292]: do you need to do this for the overdose to increase heart rate?

File: code/modules/reagents/chemistry/reagents/drugs.dm [hunk 3]
```diff
@@ -590,6 +590,68 @@
 		H.LoseBreath(10 SECONDS) // procs 5 times, mostly a visual thing. damage could stack to cause a slowdown.
 		H.Confused(10 SECONDS)
 
+/datum/reagent/happiness
+	name = "Happiness"
+	id = "happiness"
+	description = "Fills you with ecstasic numbness and causes minor brain damage. If overdosed causes sudden mood swings and spikes in heart rate."
```
Comment (JimKil3) [inline code/modules/reagents/chemistry/reagents/drugs.dm:596]:
```suggestion
	description = "Fills you with ecstasic numbness and causes minor brain damage. If overdosed, causes sudden mood swings and spikes in heart rate."
```

File: code/modules/reagents/chemistry/reagents/drugs.dm [hunk 4]
```diff
@@ -590,6 +590,68 @@
...
+	minor_addiction = TRUE
+
+/datum/reagent/happiness/on_mob_life(mob/living/M)
+	var/update_flags = STATUS_UPDATE_NONE
+	if(prob(15))
+		M.emote(pick("laugh", "giggle", "smile", "grin"))
+	else if(prob(10))
+		to_chat(M, "<span class='notice'>You feel [pick("great", "good", "amazing", "really nice", "magical")]!</span>")
+	else if(prob(1))
+		M.say("hehehe") // you WILL hehehe
```
Comment (JimKil3) [inline code/modules/reagents/chemistry/reagents/drugs.dm:615]:
```suggestion
		M.say("hehehe...") // you WILL hehehe
```
I always think it looks a little weird to say something without punctuation, but this is entirely up to you.

File: code/modules/reagents/chemistry/reagents/drugs.dm [hunk 5]
```diff
@@ -590,6 +590,68 @@
...
+
+/datum/reagent/happiness/overdose_process(mob/living/M, severity)
+	var/list/overdose_info = ..()
+	var/effect = overdose_info[REAGENT_OVERDOSE_EFFECT]
+	var/update_flags = overdose_info[REAGENT_OVERDOSE_FLAGS]
+
+	if(prob(15))
+		M.emote(pick("cry", "frown", "sulk", "gurgle"))
+	else if(prob(10))
+		to_chat(M, "<span class='warning'>You feel [pick("like shit", "terrible", "weak", "like a rhumba beat", "hollow")]!</span>")
```
Comment (JimKil3) [inline code/modules/reagents/chemistry/reagents/drugs.dm:631]:
```suggestion
		to_chat(M, "<span class='warning'>You feel [pick("like shit", "terrible", "weak", "hollow")]!</span>")
```
I would imagine feeling like a rhumba beat is pretty good? I dunno though lol

File: code/modules/reagents/chemistry/reagents/drugs.dm [hunk 6]
```diff
@@ -590,6 +590,68 @@
...
+	update_flags |= M.adjustBrainLoss(0.2, FALSE)
+	if(!prob(10 * (severity ** 2))) // 1 - 10, 2 - 40
+		return list(effect, update_flags)
+
+	var/static/list/good_messages = list("YES! YES!! YES!!", "I AM UNSTOPPABLE", "THIS IS GREAT", "THERE WILL NEVER BE ANYONE BETTER",
+		"GLORY IS MINE", "WE'RE SO BACK", "I AM FUCKING INVINCIBLE", "I'M HANGING IN THERE")
```
Comment (JimKil3) [inline code/modules/reagents/chemistry/reagents/drugs.dm:637]:
```suggestion
	var/static/list/good_messages = list("YES! YES!! YES!!", "I AM UNSTOPPABLE", "THIS IS GREAT", "THERE WILL NEVER BE ANYONE BETTER",
		"GLORY IS MINE", "WE'RE SO BACK", "I AM FUCKING INVINCIBLE", "I'M THE KING OF THE RHUMBA BEAT")
```
Making up for the earlier reference removal. Also, I don't think _just_ hanging in there fits for literally overdosing on happiness.

File: code/modules/reagents/chemistry/reagents/drugs.dm [hunk 7]
```diff
@@ -590,6 +590,68 @@
...
+	var/static/list/bad_messages = list("NO! NO!! NO!!", "OH MY GOD", "THEY'RE LOOKING AT ME", "KILLLL MEEE, KILLLLL ME",
+		"I CAN'T FUCKING TAKE IT ANYMORE", "ARRGH, IT'S OVER")
```
Comment (JimKil3) [inline code/modules/reagents/chemistry/reagents/drugs.dm:639]: Maybe add a "literally 1984" one? Might be a bit too memey though

File: code/modules/reagents/reagent_containers/pill.dm [hunk 8]
```diff
@@ -87,12 +87,23 @@
 	icon_state = "pill8"
 	list_reagents = list("haloperidol" = 15)
 
+/obj/item/reagent_containers/food/pill/happy_psych
+	name = "mood stabilizer pill"
+	desc = "Used to temporarily alleviate anxiety and depression, take only as prescribed."
```
Comment (JimKil3) [inline code/modules/reagents/reagent_containers/pill.dm:92]:
```suggestion
	desc = "Used to temporarily alleviate anxiety and depression. Take only as prescribed."
```

File: code/modules/reagents/reagent_containers/pill.dm [hunk 9]
```diff
@@ -87,12 +87,23 @@
...
+	icon_state = "pill_happy"
+	list_reagents = list("happiness" = 15, "mannitol" = 5)
+
 /obj/item/reagent_containers/food/pill/happy
 	name = "happy pill"
-	desc = "Happy happy joy joy!"
-	icon_state = "pill18"
+	desc = "They have little happy faces on them, and they smell like marker pens."
```
Comment (JimKil3) [inline code/modules/reagents/reagent_containers/pill.dm:98]:
```suggestion
	desc = "They have little happy faces on them and smell like marker pens."
```

File: code/modules/surgery/surgery_helpers.dm [hunk 10]
```diff
@@ -22,6 +22,8 @@
 		return 0.85
 	if(drunk >= 15)//a little drunk
 		return 0.85
+	if(target.reagents.has_reagent("happiness")) // fuck yeah
+		return 0.81
```
Comment (JimKil3) [inline code/modules/surgery/surgery_helpers.dm:26]: I bet someone will use this on their mentor application as an example of niche knowledge in like, 2 years if this PR gets merged

**Review by JimKil3 — CHANGES_REQUESTED** [review]
Most of my comments are just grammar nitpicks. I wish more chems had a recipe like this one does, it's really interesting. The overdose messages are great too

File: code/modules/reagents/chemistry/reagents/drugs.dm (hunk 5 above)
Comment (Contrabang) [reply code/modules/reagents/chemistry/reagents/drugs.dm:631]: I dunno, the last time I got rhumba beat'd I fuckin exploded

File: code/modules/reagents/chemistry/reagents_datum.dm (hunk 2 above)
Comment (Contrabang) [reply code/modules/reagents/chemistry/reagents_datum.dm:292]: Yea, I want it to only increase on overdose

File: code/game/objects/items/contraband.dm (hunk 1 above)
Comment (Contrabang) [reply code/game/objects/items/contraband.dm:6]: I think the comma breaks up the description of them, then it describes what they actually do

Comment (JimKil3) [reply code/game/objects/items/contraband.dm:6]: Saying "for when.." also breaks them up IMO. It's not like either way is wrong, I just think removing the comma makes it flow better.

File: code/modules/reagents/chemistry/reagents/drugs.dm (hunk 5 above)
Comment (JimKil3) [reply code/modules/reagents/chemistry/reagents/drugs.dm:631]: Now that you say that it makes more sense, but you won't be able to explain that to every single person overdosing on the chem. Up to you though, it's not like this is particularly important to the function of the drug or anything

**Comment by Fox-McCloud** [comment]
I'm disappointed that happiness is attainable.

Unrealistic and breaks immersion. 😎

**Review by lewcc — APPROVED** [review]

**Review by Burzah — APPROVED** [review]
lgtm

**Review by DGamerL — APPROVED** [review]

**Review by S34NW — APPROVED** [review]

**Pull request merged**